addons/l10n_fr_certification/models/account.py

```diff
@@ -6,7 +6,8 @@
 from openerp.tools.translate import _
 from openerp.exceptions import UserError
 
-ERR_MSG = _("According to the french law, you cannot modify a %s in order for its posted data to be updated or deleted. Field: %s")
+
+ERR_MSG = _("According to the French law, you cannot modify a %s in order for its posted data to be updated or deleted. Unauthorized field: %s.")
```

Comment:
The translation here doesn't work. The `_()` marker will let the system detect a term to include in our translation terms, but it won't actually be able to translate it, because there is no context to do it when the Python file is loaded.
You would need to translate it *also* at runtime wherever ERR_MSG is used, e.g. `_(ERR_MSG)`.

Additionally, the use of "%s" placeholders here will make translating the sentence awkward. And same thing for the terms that will fill the placeholders, that will be out of context for the translator.

As we only have a couple of variations, it would be both more robust and translator-friendly to just duplicate the whole sentence for every variation. And if the code using the messages is duplicated in many places, you can keep them as "constants", but remember you will need to translate them twice!

Same comments goes for all other occurrences of this pattern, for example `BKN_NO_TOUCH` (which has even worse placeholders IMO ;-))

Reply:
@odony interesting, I didn't know the pitfall of translating global variables... but I have to back up @kebeclibre for the %s placeholders: it's a common usage we do all over the code. :-s